scripts/west_commands/runners/pyocd.py

```diff
@@ -148,18 +148,23 @@ def do_run(self, command, **kwargs):
             self.debug_debugserver(command, **kwargs)
 
     def flash(self, **kwargs):
+        # Use hex, bin or elf file provided by the buildsystem.
+        # Preferring .hex over .bin and .elf
         if self.hex_name is not None and os.path.isfile(self.hex_name):
             fname = self.hex_name
+        # Preferring .bin over .elf
         elif self.bin_name is not None and os.path.isfile(self.bin_name):
             self.logger.warning(
                 'hex file ({}) does not exist; falling back on .bin ({}). '.
                 format(self.hex_name, self.bin_name) +
                 'Consider enabling CONFIG_BUILD_OUTPUT_HEX.')
```

Comment:
It is somewhat weird to have a warning in the `bin` case, but not for `elf`?

Reply:
> It is somewhat weird to have a warning in the `bin` case, but not for `elf`?

Agree. The weird warning has been deleted.
Other runners do not have it.